src/update.rs

```diff
@@ -288,6 +289,29 @@ where
     }
 }
 
+unsafe impl<T> Update for Arc<T>
```

Comment:
I don't feel too confident about the `Update` implementations here. I never fully understood how `Update` is supposed to work...

Reply:
Agreed, the docs on the trait are not that clear unfortunately, Niko clearly tried making them clear but its still a bit difficult to understand. From what I understand (by reading the existing implementations) we likely want to "peel" away one layer of types and delegate the `maybe_update` down?

Reply:
Yeah, that's the pattern I tried to follow. I'm not sure if it's worth doing this entire ceremony for `Arc` by trying to update them in place but it seemed reasonable to do?